src/components/shoppingCart/paymentbar/PaymentSum.jsx

```diff
@@ -0,0 +1,64 @@
+import React, { useState, useEffect } from "react";
+import Container from "react-bootstrap/Container";
+import Col from "react-bootstrap/Col";
+import Row from "react-bootstrap/Row";
+import Button from "react-bootstrap/Button";
+import "./PaymentSum.scss";
+
+// TODO(emrerdem1): To get these datas from Redux.
+const SHIPPING_COST = 10;
+const AMOUNT_TO_FREE_SHIPPING = 150;
+const TEMPORARY_SUBTOTAL = 160;
+
+const PaymentSum = () => {
+  const [shippingCost] = useState(SHIPPING_COST);
+  const [isShippingFree, setIsShippingFree] = useState(false);
+  const [subTotal] = useState(TEMPORARY_SUBTOTAL);
+  const [totalCost, setTotalCost] = useState(null);
+
+  useEffect(() => {
+    if (subTotal >= AMOUNT_TO_FREE_SHIPPING) {
+      setIsShippingFree(true);
+      setTotalCost(subTotal);
+      return;
+    } else {
+      setIsShippingFree(false);
+      setTotalCost(shippingCost + subTotal);
+    }
+  }, [subTotal, shippingCost]);
```

Comment:
so actually (and this is kind of important)

the only two things that matter are subTotal and shippingCost, from what I see. everything else is a function of the two. in that case, you don't actually use state, you do it like so
```
const subTotal = ..
const shippingCist = ...

let isShippingFree // A NORMAL VARIABLE NOT A STATE VARIABLE
if (...) {
  isShippingFree = true
  totalCost = blah
}
```

these get calculated on every rerender automatically when the state changes (shippingcost and subtotal). tldr no need for react.useeffect